File: libs/bus-core/integrations/integrations.go
package integrations

const (
	AddIntegrationTopic    = "integrations.add"
	RemoveIntegrationTopic = "integrations.remove"
)

type Service string

const (
	DonationAlerts Service = "DONATIONALERTS"
	Streamlabs     Service = "STREAMLABS"
	DonatePay      Service = "DONATEPAY"
	Faceit         Service = "FACEIT"
)

type Request struct {
	ID      string  `json:"id"`
	Service Service `json:"service"`
}

File: libs/bus-core/src/integrations/integrations.ts
export const AddIntegrationTopic = 'integrations.add'
export const RemoveIntegrationTopic = 'integrations.remove'

export enum IntegrationService {
	DONATIONALERTS = 'DONATIONALERTS',
	STREAMLABS = 'STREAMLABS',
	DONATEPAY = 'DONATEPAY',
	FACEIT = 'FACEIT',
}

export interface Request {
	id: string
	service: IntegrationService
}
